internals/secrethub/migrate.go

```diff
@@ -450,38 +434,79 @@ func (cmd *MigrateApplyCommand) Run() error {
 				return err
 			}
 		} else {
-			if !cmd.append {
-				confirmed, err := ui.AskYesNo(cmd.io, fmt.Sprintf("A vault named %s already exists. Do you want to append the items to that vault?", vault.Name), ui.DefaultYes)
-				if err != nil {
-					return fmt.Errorf("a vault named %s already exists. Use --apend to append to existing vaults", vault.Name)
-				}
-				if !confirmed {
-					fmt.Fprintln(cmd.io.Output(), "Aborting.")
-					return nil
-				}
-			}
-
-			fmt.Fprintf(cmd.io.Output(), "[%d/%d] Appending to existing vault: %s\n", i, len(plan.vaults), vault.Name)
+			fmt.Fprintf(cmd.io.Output(), "[%d/%d] Updating vault: %s\n", i, len(plan.vaults), vault.Name)
 		}
 
 		for _, item := range vault.Items {
-			template := onepassword.NewItemTemplate()
-			for _, field := range item.Fields {
-				value, err := client.Secrets().ReadString(strings.TrimPrefix(field.Reference, secretReferencePrefix))
+			exists, err := onepassword.ExistsItemInVault(vault.Name, item.Name)
+			if err != nil {
+				return err
+			}
+			if !exists {
+				template := onepassword.NewItemTemplate()
+				for _, field := range item.Fields {
+					value, err := client.Secrets().ReadString(strings.TrimPrefix(field.Reference, secretReferencePrefix))
+					if err != nil {
+						return err
+					}
+					template.AddField(field.Name, value, field.Concealed)
+				}
+
+				err = onepassword.CreateItem(vault.Name, template, item.Name)
 				if err != nil {
 					return err
 				}
-				template.AddField(field.Name, value, field.Concealed)
-			}
-
-			err = onepassword.CreateItem(vault.Name, template, item.Name)
-			if err != nil {
-				return err
+			} else {
+				for _, field := range item.Fields {
+					hasField, err := onepassword.HasField(vault.Name, item.Name, field.Name)
```

Comment:
I noticed the command is getting a bit slower. Of course that is expected now that we're checking for existence, but I believe we can make a few changes to speed it up a bit:

We're currently fetching every item twice times the number of fields of the item: once for `HasField` and once for `GetField`. Shall we instead fetch the item once, maybe returning a map of fields and values, which we re-use for all fields and both lookups? For a 4 field item, that leaves only 1 `op get item` request, instead of the current 8 requests.

Reply:
Great suggestion! I implemented it